src/htm/algorithms/Connections.cpp

```diff
@@ -433,7 +429,8 @@ void Connections::adaptSegment(const Segment segment,
                                const SDR &inputs,
                                const Permanence increment,
                                const Permanence decrement, 
-			       const bool pruneZeroSynapses)
+			       const bool pruneZeroSynapses, 
+			       const UInt segmentThreshold)
```

Comment:
these 2 options could be merged together, but I like to keep them separate: first gates synapse pruning, the latter segment pruning.